mail_restrict_follower_selection/models/mail_wizard_invite.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (C) 2015 Therp BV <http://therp.nl>
+# Copyright (C) 2017 Komit <http://www.komit-consulting.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import ast
+
+from lxml import etree
+
+from odoo import api, models
+
+
+class MailWizardInvite(models.TransientModel):
+    _inherit = "mail.wizard.invite"
+
+    @api.model
+    def _mail_restrict_follower_selection_get_domain(self, res_model=None):
+        if not res_model:
+            res_model = self.env.context.get("default_res_model")
+        parameter_name = "mail_restrict_follower_selection.domain"
+        return (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param(
+                "{}.{}".format(parameter_name, res_model),
+                self.env["ir.config_parameter"]
+                .sudo()
+                .get_param(parameter_name, default="[]"),
+            )
+        )
+
+    @api.model
+    def get_view(self, view_id=None, view_type="form", **options):
+        result = super(MailWizardInvite, self).get_view(
+            view_id=view_id, view_type=view_type, **options
+        )
+        arch = etree.fromstring(result["arch"])
+        for field in arch.xpath('//field[@name="partner_ids"]'):
+            domain = self._mail_restrict_follower_selection_get_domain()
+            domain = ast.literal_eval(domain)
+            domain.append(("type", "!=", "private"))
```

Comment:
Can we get this domain from `self._fields["partner_ids"].domain` (and use expression.AND(), perhaps)?

I think this domain manipulation should be done inside _mail_restrict_follower_selection_get_domain() instead of here.